core/src/OpenACC/Kokkos_OpenACC_Instance.hpp

```diff
@@ -46,24 +46,31 @@
 #define KOKKOS_OPENACC_INSTANCE_HPP
 
 #include <impl/Kokkos_InitializationSettings.hpp>
+#include <impl/Kokkos_Tools.hpp>
 
 #include <cstdint>
 #include <iosfwd>
 #include <string>
+#include <openacc.h>
 
 namespace Kokkos::Experimental::Impl {
 
 class OpenACCInternal {
   bool m_is_initialized = false;
 
-  OpenACCInternal()                       = default;
   OpenACCInternal(const OpenACCInternal&) = default;
   OpenACCInternal& operator=(const OpenACCInternal&) = default;
 
  public:
-  static OpenACCInternal* singleton();
+  int m_accDev;
```

Comment:
```suggestion
  static int m_accDev;
```
You might as well make it static since it is not legal to use anything else than the device selected at initialization.

Reply:
Changed.